test/boost/hashers_test.cc

```diff
@@ -102,3 +102,17 @@ SEASTAR_THREAD_TEST_CASE(mutation_fragment_sanity_check) {
         check_hash(f, 0x5092daca1b27ea26ull);
```

Comment:
I think consistent header content is more important than avoiding a circular header dependency. Those are only a problem if they are cross-modules, but that is not the case here.

Reply:
This means we'll need to implement this basic hasher in its own header that doesn't include bytes.hh.
I'll see if there's a better way around it...